python/unblob/file_utils.py

```diff
@@ -269,8 +269,15 @@ def iterate_patterns(
         file.seek(initial_position)
 
 
+class RandomReader(Protocol):
+    # File implements this interface
+
+    def read(self, n: Optional[int] = None) -> bytes: ...
+    def seek(self, pos: int, whence: int = io.SEEK_SET) -> int: ...
```

Comment:
I was thinking about getting `File` (and `OffsetFile` in sync with `io.*` instead, as they use different parameter names.
```py
read(self, size=-1, /)
seek(self, offset, whence=0, /)
```
but it turns out, `io.seek()` has inconsistent names for the first parameter: `offset`, `pos`, target`, and `cookie` anyway, so just gave up on it.